File: stock_platform/python/backend/python/services/dividend_service.py
from models import Dividend
from schemas import DividendCreate


async def create_dividend(dividend: DividendCreate) -> Dividend:
    return await Dividend.create(
        account_id=dividend.account_id,
        stock_code=dividend.stock_code,
        stock_name=dividend.stock_name,
        dividend_amount=dividend.dividend_amount,
        dividend_date=dividend.dividend_date
    )


async def list_dividends(account_id: int | None = None) -> list[Dividend]:
    query = Dividend.all()
    if account_id is not None:
        query = query.filter(account_id=account_id)
    return await query.order_by("-dividend_date")

File: stock_platform/java/backend/java/src/main/java/com/stockplatform/service/DividendService.java
package com.stockplatform.service;

import com.my.mybatis.flex.core.condition.QueryCondition;
import com.stockplatform.dto.DividendCreateDTO;
import com.stockplatform.dto.DividendResponseDTO;
import com.stockplatform.entity.Dividend;
import com.stockplatform.mapper.DividendMapper;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class DividendService {

    private final DividendMapper dividendMapper;

    public DividendService(DividendMapper dividendMapper) {
        this.dividendMapper = dividendMapper;
    }

    public DividendResponseDTO createDividend(DividendCreateDTO dto) {
        Dividend dividend = new Dividend();
        dividend.setAccountId(dto.getAccountId());
        dividend.setStockCode(dto.getStockCode());
        dividend.setStockName(dto.getStockName());
        dividend.setDividendAmount(dto.getDividendAmount());
        dividend.setDividendDate(dto.getDividendDate());
        dividend.setCreatedAt(LocalDateTime.now());
        dividendMapper.insertSelective(dividend);
        return DividendResponseDTO.fromEntity(dividend);
    }

    public List<DividendResponseDTO> listDividends(Integer accountId) {
        QueryCondition condition = QueryCondition.create();
        if (accountId != null) {
            condition.where("account_id = ?", accountId);
        }
        condition.orderBy("dividend_date", false);
        return dividendMapper.selectAllByCondition(condition).stream()
                .map(DividendResponseDTO::fromEntity)
                .collect(Collectors.toList());
    }
}